File: C_Programming/binary_converter.c
//Input: Decimal Number (Integer -32768 to 65535)
//Output: Binary Number (Signed 2's Complement)

#include <stdio.h>

int main( )
{
    int num;
    printf("Enter the Decimal Number: ");
    scanf ("%d", &num);
    printf("Binary Number: ");
    
    for (int i = 0; i < 16; i++)
        printf ("%d", (num << i & 1 << 15 ) ? 1 : 0);

    return 0;
}

File: Python/binary_converter.py
#Input: Decimal Number (Integer)
#Output: Binary Number (Signed 2's Complement)

num = int(input("Enter the Decimal Number: "))
byt = int(input("Enter the number of bytes you want to your integer be in: "))
bit = byt*8

for i in range(bit):
	if (num<<i) & (1<<(bit-1)):
		print(1, end="")
	else:
		print(0, end="")
print()

